### PR TITLE
Check that requirements.txt has all dependencies

### PR DESCRIPTION
**Pull request by dmage**
It could help avoid #1428 (and maybe #1429).

### CODE REVIEW

**Comment by HammerMeetNail** [comment]
This is interesting! I'd been using `pip show` to try and find the deps before adding to `requirements.txt`, but it didn't show any build deps. Even after install of `azure-core`, there was still no reference in `pip freeze` or anything to `flit_core`.

**Comment by dmage** [comment]
I excluded pip (`s/^pip==.*/pip/`) because it stays 22.0.4 despite requirements.txt

**Comment by HammerMeetNail** [comment]
@dmage How would this have caught `flit-core`? I think it's cross checking against `pip freeze`, but `flit-core` comes as part of `typing-extensions` and doesn't show in `pip freeze`.

File: requirements.txt [hunk 1]
```diff
@@ -143,8 +143,8 @@ zope.event==4.5.0
 zope.interface==5.4.0
 
 # Build dependencies
-Cython==3.0a9
-flit_core==3.7.1
+Cython==3.0.0a9
+flit-core==3.7.1
```
Comment (HammerMeetNail) [inline requirements.txt:147]: Is this right? I'd confirmed that `pip install flit_core==3.7.1` installs. https://pypi.org/project/flit_core/

**Review by HammerMeetNail — APPROVED** [review]
LGTM 👍

**Comment by dmage** [comment]
We've got a new version of `pip` in master, so I updated this PR.

**Review by HammerMeetNail — APPROVED** [review]
LGTM 👍

**Pull request merged**